### PR TITLE
Fix datafusion-cli publishing

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of #9682 

## Rationale for this change

I had to add a `version` field to the datafusion-cli `Cargo.toml` file to correctly publish to crates.io as `datafusion-common` did not specify a version

## What changes are included in this PR?
1. Remove direct datafuson_common  and use existing `datafusion` dependency

## Are these changes tested?
By C

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion-cli/src/command.rs [hunk 1]
```diff
@@ -26,9 +26,9 @@ use datafusion::arrow::array::{ArrayRef, StringArray};
 use datafusion::arrow::datatypes::{DataType, Field, Schema};
 use datafusion::arrow::record_batch::RecordBatch;
 use datafusion::common::exec_err;
+use datafusion::common::instant::Instant;
```
Comment (alamb) [inline datafusion-cli/src/command.rs:29]: the whole change is `datafusion_common` --> `datafusion::common`

File: datafusion-cli/Cargo.toml [hunk 2]
```diff
@@ -45,7 +45,6 @@ datafusion = { path = "../datafusion/core", version = "37.0.0", features = [
     "unicode_expressions",
     "compression",
 ] }
-datafusion-common = { path = "../datafusion/common" }
```
Comment (alamb) [inline datafusion-cli/Cargo.toml:48]: I had to change this to
```toml
datafusion-common = { path = "../datafusion/common", version="37.0.0" }
```

to get `datafusion-cli` to publish. I think removing the whole dependency is cleaner that adding a version that needs to be kept in sync

**Review by Jefffrey — APPROVED** [review]
:+1:

**Comment by alamb** [comment]
Thank you @Jefffrey

**Pull request merged**